scrapy/loader/__init__.py

```diff
@@ -3,219 +3,104 @@
+        proc = self._get_item_field_attr(field_name, 'output_processor')
         if not proc:
-            proc = self._get_item_field_attr(field_name, 'output_processor',
-                                             self.default_output_processor)
-        return unbound_method(proc)
-
-    def _process_input_value(self, field_name, value):
-        proc = self.get_input_processor(field_name)
-        _proc = proc
-        proc = wrap_loader_context(proc, self.context)
-        try:
-            return proc(value)
-        except Exception as e:
-            raise ValueError(
-                "Error with input processor %s: field=%r value=%r "
-                "error='%s: %s'" % (_proc.__class__.__name__, field_name,
-                                    value, type(e).__name__, str(e)))
-
-    def _get_item_field_attr(self, field_name, key, default=None):
+            proc = super().get_default_output_processor_for_field(field_name)
+        return proc
+
+    def _get_item_field_attr(self, field_name, key):
         if isinstance(self.item, Item):
-            value = self.item.fields[field_name].get(key, default)
-        else:
-            value = default
-        return value
-
-    def _check_selector_method(self):
-        if self.selector is None:
-            raise RuntimeError("To use XPath or CSS selectors, "
-                               "%s must be instantiated with a selector "
-                               "or a response" % self.__class__.__name__)
-
-    def add_xpath(self, field_name, xpath, *processors, **kw):
-        values = self._get_xpathvalues(xpath, **kw)
-        self.add_value(field_name, values, *processors, **kw)
-
-    def replace_xpath(self, field_name, xpath, *processors, **kw):
-        values = self._get_xpathvalues(xpath, **kw)
-        self.replace_value(field_name, values, *processors, **kw)
-
-    def get_xpath(self, xpath, *processors, **kw):
-        values = self._get_xpathvalues(xpath, **kw)
-        return self.get_value(values, *processors, **kw)
-
-    def _get_xpathvalues(self, xpaths, **kw):
-        self._check_selector_method()
-        xpaths = arg_to_iter(xpaths)
-        return flatten(self.selector.xpath(xpath).getall() for xpath in xpaths)
-
-    def add_css(self, field_name, css, *processors, **kw):
-        values = self._get_cssvalues(css, **kw)
-        self.add_value(field_name, values, *processors, **kw)
-
-    def replace_css(self, field_name, css, *processors, **kw):
-        values = self._get_cssvalues(css, **kw)
-        self.replace_value(field_name, values, *processors, **kw)
-
-    def get_css(self, css, *processors, **kw):
-        values = self._get_cssvalues(css, **kw)
-        return self.get_value(values, *processors, **kw)
-
-    def _get_cssvalues(self, csss, **kw):
-        self._check_selector_method()
-        csss = arg_to_iter(csss)
-        return flatten(self.selector.css(css).getall() for css in csss)
+            return self.item.fields[field_name].get(key)
```

Comment:
I guess we need to wait for #3881 and use `ItemAdapter` here. Maybe also in the `itemloaders` library itself?

Reply:
That's the question, what should we merge first :)

Reply:
Both work for me.
Just need to keep in mind that, as we wait, new changes may come to the files that were moved, requiring to patch/merge changes in the new repo..
This can get a bit complicated, as we merged the changes there (so we may need to go back in history to merge and then apply the changes back again)